Parking planner integration

AadarshHegde123 opened the pull request:
This PR builds atop the recently merged Parking + Perception PR: https://github.com/krishauser/GEMstack/pull/174 

We add our planning logic on top of this PR to support the full end to end pipeline of perpendicular parking.

Commit:
adding everything required for planning integration

Commit:
fix for simulator

Commit:
renaming parking goal for clarity + adding on board launch functionality

Commit:
updating launch

Commit:
revert cone_detection to avoid merge conflicts

Commit:
comment + print statement clean up

krishauser reviewed (changes requested):
Can you explain why you took out goal and added parking_goal? This seems to conflict with summoning / inspection

AadarshHegde123 commented:
> Can you explain why you took out goal and added parking_goal? This seems to conflict with summoning / inspection

Oh, I did not realize they also make use of the same goal variable. The rename was just for clarity on our end. I can change this back.

Commit:
reverting back to "goal"

AadarshHegde123 commented:
Changed parking_goal back to goal to avoid conflicts.

Commit:
small updates to showcase scanning functionality in simulation

AadarshHegde123 commented:
@krishauser All good on our end for re-review. I see that there are a couple merge conflicts so I will note the differences and give explanations here:

gem_e4_dynamics.yaml: The gem e4 dynamics model we chose is selected because it prevents some obscure behavior in both simulator and on-board for our tests.

longitudinal_planning: The code is largely the same, but we have refactored such that hard-coding has been eliminated so you can now specify different arguments via the launch file.

route_planning_component.py: This conflict seemed pretty trivial since it was mainly complaining about a difference in imports and some helper functions from inspection team that were missing in our file. I fixed this in the web editor and marked as resolved.

krishauser commented on GEMstack/knowledge/defaults/current.yaml:
Was this cleared with the control team?

Jugthegreat replied:
Control Team is using kris model so this may create some conflicts. I am talking with parking team to sort this out

AadarshHegde123 replied:
I spoke with Jugal. We had an out outdated pid_p value that we have now updated to match Controls. We do however need crosstrack_gain set to 0.5 and @Jugthegreat  has agreed that this is an acceptable change.

krishauser commented on GEMstack/knowledge/vehicle/gem_e4_dynamics.yaml:
Was this cleared with the control team?

Jugthegreat replied:
Control Team is using kris model so this may create some conflicts. I am talking with parking team to sort this out

AadarshHegde123 replied:
It seemed we had an outdated pid_p value that seemed to affect the kris model. We have re-tested with the kris_v1 model and performance is as expected. We have changed this file to match control team, so there should no longer be any issue.

krishauser commented on GEMstack/onboard/planning/pure_pursuit.py:
Did you clear this with the control team?

AadarshHegde123 replied:
Yes, I spoke with Jugal and he has said that Controls is fine with the changes made here.

krishauser reviewed (changes requested):
Need coordination with control team. Either @Jugthegreat can comment or message me to approve these changes.

Commit:
updating controls

Commit:
fixing a conflict

AadarshHegde123 commented:
@krishauser This should be ready again for re-review, we have talked everything over with controls team. In regards to the merge conflicts, I have resolved resolve parking_detection.py and route_planning_component.py, but the existing longitudinal_planning.py on s2025 from my tests seems non-functional (see attached video).

I propose having two separate motion_planning files, the current existing one on s2025 since I am guessing this is what worked for inspection team if they merged it, and this parking_motion_planning file that I have created. These files have diverged significantly to the point where I think its reasonable to have them separate. I have made this change to this branch.

Let me know if you have any concerns with this.

https://github.com/user-attachments/assets/a7c49ce6-947e-4425-a507-012f0d2bb81f

Commit:
Merge branch 's2025' into parking-planning-merge

Commit:
resolving issues on s2025

krishauser reviewed (approved):
LGTM
